lvgl_touch/gt911.c

```diff
@@ -0,0 +1,168 @@
+
+    i2c_master_read_byte(i2c_cmd, data_buf, I2C_MASTER_NACK);
+    i2c_master_stop(i2c_cmd);
+    esp_err_t ret = i2c_master_cmd_begin(TOUCH_I2C_PORT, i2c_cmd, 1000 / portTICK_RATE_MS);
+    i2c_cmd_link_delete(i2c_cmd);
+    return ret;
+}
+
+esp_err_t gt911_i2c_write8(uint8_t slave_addr, uint16_t register_addr, uint8_t data) {
+    i2c_cmd_handle_t i2c_cmd = i2c_cmd_link_create();
+
+    i2c_master_start(i2c_cmd);
+    i2c_master_write_byte(i2c_cmd, (slave_addr << 1) | I2C_MASTER_WRITE, true);
+    i2c_master_write_byte(i2c_cmd, (register_addr >> 8), I2C_MASTER_ACK);
+    i2c_master_write_byte(i2c_cmd, (register_addr & 0xFF), I2C_MASTER_ACK);
+    i2c_master_write_byte(i2c_cmd, data, I2C_MASTER_ACK);
+    i2c_master_stop(i2c_cmd);
+    esp_err_t ret = i2c_master_cmd_begin(TOUCH_I2C_PORT, i2c_cmd, 1000 / portTICK_RATE_MS);
+    i2c_cmd_link_delete(i2c_cmd);
+    return ret;
+}
+
+/**
+  * @brief  Initialize for GT911 communication via I2C
+  * @param  dev_addr: Device address on communication Bus (I2C slave address of GT911).
+  * @retval None
+  */
+void gt911_init(uint8_t dev_addr) {
+    if (!gt911_status.inited) {
+        gt911_status.i2c_dev_addr = dev_addr;
+        uint8_t data_buf;
+        esp_err_t ret;
+
+        ESP_LOGI(TAG, "Checking for GT911 Touch Controller");
+        if ((ret = gt911_i2c_read(dev_addr, GT911_PRODUCT_ID1, &data_buf, 1) != ESP_OK)) {
+            ESP_LOGE(TAG, "Error reading from device: %s",
+                        esp_err_to_name(ret));    // Only show error the first time
+            return;
+        }
+
+        // Read 4 bytes for Product ID in ASCII
+        for (int i = 0; i < GT911_PRODUCT_ID_LEN; i++) {
+            gt911_i2c_read(dev_addr, (GT911_PRODUCT_ID1 + i), (uint8_t *)&(gt911_status.product_id[i]), 1);
+        }
+        ESP_LOGI(TAG, "\tProduct ID: %s", gt911_status.product_id);
+
+        gt911_i2c_read(dev_addr, GT911_VENDOR_ID, &data_buf, 1);
+        ESP_LOGI(TAG, "\tVendor ID: 0x%02x", data_buf);
+
+        gt911_i2c_read(dev_addr, GT911_X_COORD_RES_L, &data_buf, 1);
+        gt911_status.max_x_coord = data_buf;
+        gt911_i2c_read(dev_addr, GT911_X_COORD_RES_H, &data_buf, 1);
+        gt911_status.max_x_coord |= ((uint16_t)data_buf << 8);
+        ESP_LOGI(TAG, "\tX Resolution: %d", gt911_status.max_x_coord);
+
+        gt911_i2c_read(dev_addr, GT911_Y_COORD_RES_L, &data_buf, 1);
+        gt911_status.max_y_coord = data_buf;
+        gt911_i2c_read(dev_addr, GT911_Y_COORD_RES_H, &data_buf, 1);
+        gt911_status.max_y_coord |= ((uint16_t)data_buf << 8);
+        ESP_LOGI(TAG, "\tY Resolution: %d", gt911_status.max_y_coord);
+        gt911_status.inited = true;
+    }
+}
+
+/**
+  * @brief  Get the touch screen X and Y positions values. Ignores multi touch
+  * @param  drv:
+  * @param  data: Store data here
+  * @retval Always false
+  */
+bool gt911_read(lv_indev_drv_t *drv, lv_indev_data_t *data) {

```

Comment:
Should return `void`.

Reply:
I don't think it's needed; the translation to the v8 approach is done [in `touch_driver_read`](https://github.com/lvgl/lvgl_esp32_drivers/pull/69/files#diff-ecabe01db47fce3a7d73d2e391e5ade59d1560f403097408668c9d2317e976c1R49-R56).

Reply:
Does it mean the indev driver in this repo will keep following v7 API?

Reply:
The top level `read_cb` function (`touch_driver_read`) uses the v8 API, however, to avoid needing to refactor all the individual drivers, they are still using the v7 API.

Reply:
What about defining a symbol like `INDEV_READ_RETURN_TYPE` and use it like this `INDEV_READ_RETURN_TYPE gt911_read(lv_indev_drv_t *drv, lv_indev_data_t *data)`. We could define it to `void` when using v8, and `bool` when using v7.

Reply:
> What about defining a symbol like INDEV_READ_RETURN_TYPE and use it like this INDEV_READ_RETURN_TYPE gt911_read(lv_indev_drv_t *drv, lv_indev_data_t *data). We could define it to void when using v8, and bool when using v7.

This way the last `return false;` still needs to be `#if`ed. 

I think it's not a problem if the `bool` return value is kept if it makes the handling of v7 easier. I just wanted to be sure it's intentional. :slightly_smiling_face: